scripts/Interface/ui/drill/presenter/DrillPresenter.py

```diff
@@ -513,73 +399,76 @@ def onClose(self):
-            self._saveDataQuestion()
+            return self._saveDataQuestion()
+        return True
 
     def onNew(self):
         """
         Triggered when the user wants an empty table.
         """
         if self.view.isWindowModified():
-            self._saveDataQuestion()
+            if not self._saveDataQuestion():
+                return
         self.model.clear()
         self.model.resetIOFile()
         self.view.setWindowTitle("Untitled [*]")
         self._syncViewHeader()
-        self._syncViewTable()
+        if self._resetTable():
+            self.model.addSample(0)
 
     def _saveDataQuestion(self):
         """
         Open a dialog to ask the user if he wants to save the rundex file.
+
+        Returns:
+            bool: True if the user decided to continue
         """
         if self.view.isHidden():
             return
         q = QMessageBox.question(self.view, "DrILL: Unsaved data", "You have "
                                  "unsaved modifications, do you want to save "
                                  "them before?",
-                                 QMessageBox.Yes | QMessageBox.No)
-        if (q == QMessageBox.Yes):
+                                 QMessageBox.Yes | QMessageBox.No
+                                 | QMessageBox.Cancel)
+        if q == QMessageBox.Yes:
             self.onSaveAs()
+            return True
+        if q == QMessageBox.No:
+            return True
+        if q == QMessageBox.Cancel:
+            return False
 
     def _syncViewHeader(self):
+        """
+        Synchronize the drill header with the model.
+        """
         availableModes = self.model.getAvailableAcquisitionModes()
+        instrument = self.model.getInstrument()
         acquisitionMode = self.model.getAcquisitionMode()
         cycle, exp = self.model.getCycleAndExperiment()
 
+        self.view.blockSignals(True)
+        self.view.setInstrument(instrument)
         self.view.set_available_modes(availableModes)
         self.view.set_acquisition_mode(acquisitionMode)
         self.view.setCycleAndExperiment(cycle, exp)
+        self.view.blockSignals(False)
 
-    def _syncViewTable(self):
-        columns, tooltips = self.model.getColumnHeaderData()
-        if tooltips and tooltips[-1] == "CustomOptions":
-            tooltips[-1] = "Provide semicolon (;) separated key=value pairs"
-        samples = self.model.getSamples()
-        groups = self.model.getSamplesGroups()
-        masters = self.model.getMasterSamples()
-
-        self.view.blockSignals(True)
+    def _resetTable(self):
+        """
+        Reset the table header.
+        """
+        acquisitionMode = self.model.getAcquisitionMode()
+        if acquisitionMode not in RundexSettings.COLUMNS:
+            self.view.set_table([], [])
+            return False
+        parameters = self.model.getParameters()
+        columns = RundexSettings.COLUMNS[acquisitionMode]
+        tooltips = list()
+        for name in columns:
```

Comment:
This looks redundant. Can one param match more than one column? if not break. Again, you can have a small function that returns the doc of a parameter by name.